File: packages/database/seed_penta_brain.ts
import { PrismaClient } from '@prisma/client';
import { v4 as uuidv4 } from 'uuid';

const prisma = new PrismaClient();

async function main() {
  console.log('Cleaning up old test data...');
  await prisma.hermesAgentTask.deleteMany({});
  await prisma.actionProposal.deleteMany({});
  await prisma.funnelEvent.deleteMany({});
  await prisma.lead.deleteMany({});
  await prisma.campaign.deleteMany({});

  console.log('Seeding Campaigns...');
  const campaign = await prisma.campaign.create({
    data: {
      name: 'Summer Launch 2026',
      sourceType: 'PAID_AD',
      platform: 'meta',
      budgetCents: 50000,
      spendCents: 12500,
    },
  });

  console.log('Seeding Leads, FunnelEvents, HermesTasks & VaultProposals...');
  const stages = ['VISITOR', 'LEAD', 'MQL', 'SQL', 'CUSTOMER'] as const;
  const statuses = ['PENDING', 'RUNNING', 'COMPLETED', 'STUCK', 'AWAITING_APPROVAL'] as const;

  for (let i = 0; i < 15; i++) {
    const stage = stages[Math.floor(Math.random() * stages.length)];
    const lead = await prisma.lead.create({
      data: {
        email: `test_lead_${i}_${uuidv4().substring(0, 6)}@example.com`,
        firstName: `Test${i}`,
        lastName: 'User',
        funnelStage: stage,
        leadScore: Math.floor(Math.random() * 80) + 10,
        campaignId: campaign.id,
      },
    });

    const event = await prisma.funnelEvent.create({
      data: {
        leadId: lead.id,
        eventType: 'lead_created',
        metadata: { ip: '127.0.0.1', browser: 'Chrome' },
      },
    });

    const status = statuses[Math.floor(Math.random() * statuses.length)];
    let proposalId: string | null = null;
    let agentKey = 'lead_gen_v1';

    if (status === 'AWAITING_APPROVAL') {
      agentKey = 'copywriter_v1';
      const prop = await prisma.actionProposal.create({
        data: {
          agentId: agentKey,
          agentName: 'Copywriter Agent',
          category: 'BRAND_CONTENT_PUBLISH',
          proposedPayload: { draft: `Check out our new offer, lead ${lead.id}!` },
          riskAmountCents: 0,
          justification: 'A/B test variation required based on low conversion rate.',
          status: 'PENDING',
        },
      });
      proposalId = prop.id;
    } else if (status === 'STUCK') {
      agentKey = 'lead_gen_v1';
    }

    await prisma.hermesAgentTask.create({
      data: {
        triggerEventId: event.id,
        leadId: lead.id,
        agentKey: agentKey,
        payload: { action: 'nurture_sequence_1' },
        status: status,
        vaultProposalId: proposalId,
        lastError: status === 'STUCK' ? 'API Timeout Error' : null,
        attempts: status === 'STUCK' ? 3 : 0,
      },
    });
  }

  console.log('Seed complete! 15 Leads with events, tasks, and proposals generated across clusters.');
}

main()
  .catch((e) => {
    console.error(e);
    process.exit(1);
  })
  .finally(async () => {
    await prisma.$disconnect();
  });

File: apps/backend-os/seed_penta_brain.py
"""
seed_penta_brain.py
===================
Generates cross-cluster seed data to verify Sentinel Dashboard and System Bus.
Creates:
 - Campaigns (Orion)
 - Leads + FunnelEvents (Orion)
 - ActionProposals (Vault)
 - HermesAgentTasks (Hermes)
"""
import asyncio
import os
import random
import uuid
from datetime import datetime, timedelta, timezone
from prisma import Prisma

# We use the master database URL since locally we are running a single PostgreSQL instance
db = Prisma(datasource={"url": os.getenv("DATABASE_URL")})

async def main():
    await db.connect()
    try:
        print("Cleaning up old test data...")
        await db.hermesagenttask.delete_many()
        await db.actionproposal.delete_many()
        await db.funnelevent.delete_many()
        await db.lead.delete_many()
        await db.campaign.delete_many()
        
        print("Seeding Campaigns...")
        c1 = await db.campaign.create(
            data={
                "name": "Summer Launch 2026",
                "sourceType": "PAID_AD",
                "platform": "meta",
                "budgetCents": 50000,
                "spendCents": 12500
            }
        )
        
        print("Seeding Leads, FunnelEvents, HermesTasks & VaultProposals...")
        stages = ["VISITOR", "LEAD", "MQL", "SQL", "CUSTOMER"]
        statuses = ["PENDING", "RUNNING", "COMPLETED", "STUCK", "AWAITING_APPROVAL"]
        
        for i in range(15):
            stage = random.choice(stages)
            lead = await db.lead.create(
                data={
                    "email": f"test_lead_{i}_{uuid.uuid4().hex[:6]}@example.com",
                    "firstName": f"Test{i}",
                    "lastName": "User",
                    "funnelStage": stage,
                    "leadScore": random.randint(10, 90),
                    "campaignId": c1.id,
                }
            )
            
            # Create a FunnelEvent
            event = await db.funnelevent.create(
                data={
                    "leadId": lead.id,
                    "eventType": "lead_created",
                    "metadata": {"ip": "127.0.0.1", "browser": "Chrome"}
                }
            )
            
            # Create an Agent Task in Hermes for this event
            status = random.choice(statuses)
            
            proposal_id = None
            agent_key = "lead_gen_v1"
            
            if status == "AWAITING_APPROVAL":
                agent_key = "copywriter_v1"
                # Create a Vault ActionProposal first
                prop = await db.actionproposal.create(
                    data={
                        "agentId": agent_key,
                        "agentName": "Copywriter Agent",
                        "category": "BRAND_CONTENT_PUBLISH",
                        "proposedPayload": {"draft": f"Check out our new offer, lead {lead.id}!"},
                        "riskAmountCents": 0,
                        "justification": "A/B test variation required based on low conversion rate.",
                        "status": "PENDING"
                    }
                )
                proposal_id = prop.id
            elif status == "STUCK":
                agent_key = "lead_gen_v1"
                # Simulating a failed api call
                pass
                
            await db.hermesagenttask.create(
                data={
                    "triggerEventId": event.id,
                    "leadId": lead.id,
                    "agentKey": agent_key,
                    "payload": {"action": "nurture_sequence_1"},
                    "status": status,
                    "vaultProposalId": proposal_id,
                    "lastError": "API Timeout Error" if status == "STUCK" else None,
                    "attempts": 3 if status == "STUCK" else 0
                }
            )
            
        print("Seed complete! 15 Leads with events, tasks, and proposals generated across clusters.")
    except Exception as e:
        print(f"Error seeding database: {e}")
    finally:
        await db.disconnect()

if __name__ == "__main__":
    asyncio.run(main())
